lms/djangoapps/verify_student/views.py

```diff
@@ -493,6 +488,13 @@ def _redirect_if_necessary(
             else:
                 url = reverse('verify_student_start_flow', kwargs=course_kwargs)
 
+        if message in [self.FIRST_TIME_VERIFY_MSG, self.UPGRADE_MSG] and user.is_active:
+            # If the user is active and ecommerce service is enabled redirect the user to
+            # the ecommerce checkout page.
+            ecommerce_service = EcommerceService()
+            if ecommerce_service.is_enabled(user):
+                url = ecommerce_service.checkout_page_url(sku)
```

Comment:
Actually, nevermind. That will not work. You still need to check if they have already paid.